ppl-spark-integration/src/main/java/org/opensearch/sql/common/geospatial/GeoIpCache.java

```diff
@@ -0,0 +1,31 @@
+package org.opensearch.sql.common.geospatial;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheLoader;
+
+// TODO: LoaderCache
+
+import java.util.concurrent.TimeUnit;
+
+public class GeoIpCache {
+
+    public Cache<String, CidrGeoMap> cache;
+
+    private static GeoIpCache cacheInstance = null;

```

Comment:
This should be final.
The singleton pattern can be simplified in Java to have no getInstance() method:
```suggestion
    private static GeoIpCache cacheInstance = null;
    private static final GeoIpCache INSTANCE = new GeoIpCache();
```